RockPaperScissors/RockPaperScissors/RockScissorPapaerFunction.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  rockScissorPapaerFunction.swift
+//  RockPaperScissors
+//
+//  Created by 천승현 on 2022/12/28.
+//
+
+var isGameEnd = false
+var previousWinner: PreviousType = .user
+
+func playRockScissorsPaper() {
+    while !isGameEnd {
+        var computerChoice = createRandomRockScissorsPaper()
+        printRockScissorsPaper()
+        
+        let rockScissorsPaperResult = choiceRockScissorsPaper(errorType: .invalidRockScissorsPaper)
+        switch rockScissorsPaperResult {
+        case .success(let userNumber):
+            if isGameEnd(userNumber: userNumber) {
+                print("게임 종료")
+                isGameEnd = true
+                break
+            }
+            
+            var userChoice = convertChoiceToRockScissorsPaper(userChoice: userNumber)
+            let userScenario = checkRockScissorsPaperResult(userChoice: userChoice, with: computerChoice)
+            if userScenario == .draw {
+                break
+            }
+            
+            previousWinner = checkWinner(scenario: userScenario)
+            
```

Comment:
```suggestion
            if isGameEnd(userNumber: userNumber) {
                print("게임 종료")
                isGameEnd = true
                break
            }
            
            var userChoice = convertChoiceToRockScissorsPaper(userChoice: userNumber)
            let userScenario = checkRockScissorsPaperResult(userChoice: userChoice, with: computerChoice)
            if userScenario == .draw {
                break
            }
            
            previousWinner = checkWinner(scenario: userScenario)
            
```

해당 부분을 함수로 묶어서 가독성을 높이고 싶었지만, 오히려 코드의 흐름을 방해한다고 판단되어 수정하지 않았습니다. 혹시 함수로 묶어주는 것이 더 좋았을까요?

Reply:
저도 묵찌빠게임을 구현할때 고민했던 부분인데요
지금 함수 내부에서도 분리된 함수를 호출하여 사용하고 계시고 분리해도 코드의 흐름상 하나의 큰 단위가 분리되긴 어려울것 같아서 이대로 가도 될것같습니다ㅎㅎ